src/app/components/chat/chat-message/chat-message.component.ts

```diff
@@ -1,15 +1,52 @@
 import { Component, OnInit, Input } from '@angular/core';
 import { MessageI } from '@models/message';
+import { UserService } from '@services/user.service';
+import { AuthService } from '@services/auth.service';
+import { UserI } from '@models/user';
+import { Subject, BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { takeUntil, map } from 'rxjs/operators';
 
 @Component({
   selector: 'gtm-chat-message',
   templateUrl: './chat-message.component.html',
   styleUrls: ['./chat-message.component.scss'],
 })
 export class ChatMessageComponent implements OnInit {
+  private unsubscribe$: Subject<void> = new Subject();
   @Input() chatMessage: MessageI;
 
-  constructor() {}
+  user$ = new BehaviorSubject<UserI>(null);
+  avatarUrl$ = new BehaviorSubject<string>('assets/icons/square_icon.svg');
+  loggedInUid: string;
 
-  ngOnInit(): void {}
+  constructor(private userSvc: UserService, private authSvc: AuthService) {}
+
+  ngOnDestroy(): void {
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
+  }
+
+  ngOnInit(): void {
+    this.authSvc.authInfo$.subscribe(info => (this.loggedInUid = info.uid));
+
+    const userRef$ = this.getUserObservable(this.chatMessage.senderId);
+    userRef$.subscribe(user => {
+      this.user$.next(user);
+      const uid = this.user$.value.uid;
+      this.userSvc
+        .getAvatarUrl(uid)
+        .subscribe(url => this.avatarUrl$.next(url));
+    });
```

Comment:
I wasn't expecting you to use this operator, but I'll show you and we'll update it to this before merging it in.
```suggestion
    const user$ = this.getUserObservable(this.chatMessage.senderId);
    const avatarUrl$ = this.userSvc
      .getAvatarUrl(this.chatMessage.senderId)
      .pipe(takeUntil(this.unsubscribe$));

    combineLatest([user$, avatarUrl$]).subscribe(([user, avatarUrl]) => {
      this.user$.next(user);
      this.avatarUrl$.next(avatarUrl);
    });
```